### PR TITLE
Example: Golden Gate protocol

### LINKED ISSUE #43: Add a helper function that combines the functionality of both add_output and use_value
It might be nice to add a helper function that combines the functionality of both add_output and use_value

_Originally posted by @jakebeal in https://github.com/SD2E/paml/pull/42#discussion_r703554671_

### PR DESCRIPTION
**Pull request by jakebeal**
Adds a first draft of a GoldenGate assembly protocol based on NEB's kit. Also resolves #43

### CODE REVIEW

File: uml/__init__.py [hunk 1]
```diff
@@ -349,6 +350,41 @@ def activity_final(self):
...
+    :param param_type: type of value expected for the output
+    :param source: ActivityNode whose ObjectValue output should be routed to the source
+    :return: ActivityParameterNode associated with the output
+    """
+    parameter = self.add_output(name=name, param_type=param_type)
+    node = ActivityParameterNode(parameter=parameter.property_value)
+    self.nodes.append(node)
+    self.use_value(source, node)
+    return node
+Activity.designate_output = activity_designate_output  # Add to class via monkey patch
```
Comment (danbryce) [inline uml/__init__.py:385]: #43 addressed by this.

Comment (danbryce) [reply uml/__init__.py:385]: @jakebeal I don't see where the ActivityParameterNodes are defined?  They aren't in your mods to the uml.ttl.

Comment (danbryce) [reply uml/__init__.py:385]: Nevermind.  I see it defined in the branch, but for some reason this review has only one commit's changes (?).

**Review by danbryce — APPROVED** [review]

File: uml/__init__.py (hunk 1 above)
Comment (jakebeal) [reply uml/__init__.py:385]: It might have given you an odd view of it since I merged your own recently merged material in...

**Pull request merged**